### DISCUSSION TITLE
Nested state machine running in parallel has onDone issue

### DISCUSSION THREAD

**Discussion by thaliapill** [statelyai/xstate#3360] [General]
Disclaimer: Completely new to xstate 
xstate version: 4.29 (tried upgrading to 4.32)

I have a child state machine which has parallel nodes. This state machine gets used by a number of different machines. I was getting the error `done.invoke.<machine_name>.loading.<src_name>.fetching.invocation[0]` even though there wasn't an error which was rejected. 

After traversing the Github issues and xstate docs, there was [mention](https://github.com/statelyai/xstate/discussions/3176
) that if I upgrade to 4.32 and use an id on the machines, it could potentially rectify my issue. I have done so but I am still experiencing this issue. Do you know what else I could do to mitigate it? My onDone action for the affected machine could be the problem

```
onDone: {
     target: 'success',
     actions: assign({
                [resource]: (
                         _context: TContext,
                          event: DoneInvokeEvent<any>
                  ) => event.data,
             }) as Action<TContext, DoneInvokeEvent<any>>,
 },
```

I will work on a code sandbox now to show evidence of my issue

**Comment by Andarist** [comment]
> I was getting the error done.invoke.<machine_name>.loading.<src_name>.fetching.invocation[0]

This doesn't look like an error event so it's hard for me to build a correct mental model for what you are experiencing. 

> I will work on a code sandbox now to show evidence of my issue

This would be best cause based on such I could probably answer your question quickly. Without repros, it's really hard for maintainers to give any concrete advice as usually there are just too many unknowns.
